src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -233,42 +237,10 @@ static bool HasStateFlag(Http2OutputProducer.State state, Http2OutputProducer.St
                     // When the window update is sent, the producer will be re-queued if needed.
                     else if (hasMoreData && !aborted && !waitingForWindowUpdates)
```

Comment:
Can we simplify this by removing `waitingForWindowUpdates` since `TryScheduleNextWrite` will check this independently anyway? `waitingForWindowUpdates` only applies to stream window updates, right?

Reply:
waitingForWindowUpdates applies to both connection and stream window currently.